src/filesplugin.js

```diff
@@ -29,160 +50,155 @@ function openChannelSelector(files) {
-		 *
-		 * @param {object} fileList the one from attach()
-		 * @param {string} fileIdsStr list of files to send
-		 * @param {string} currentDir path to the current dir
-		 */
-		sendFileIdsAfterOAuth: (fileList, fileIdsStr, currentDir) => {
-			if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, fileIdsStr, currentDir', fileIdsStr, currentDir)
-			// this is only true after an OAuth connection initated from a file action
-			if (fileIdsStr) {
-				// trick to make sure the file list is loaded (didn't find an event or a good alternative)
-				// force=true to make sure we get a promise
-				fileList.changeDirectory(currentDir, true, true).then(() => {
-					const fileIds = fileIdsStr.split(',')
-					const files = fileIds.map((fid) => {
-						const f = fileList.files.find((e) => e.id === parseInt(fid))
-						if (f) {
-							return {
-								id: f.id,
-								name: f.name,
-								type: f.type,
-								size: f.size,
-							}
-						}
-						return null
-					}).filter((e) => e !== null)
-					if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, after changeDirectory, files:', files)
-					if (files.length) {
-						if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, after changeDirectory, call openChannelSelector')
-						openChannelSelector(files)
-					}
-				})
 			}
-		},
-
-		connectToMattermost: (selectedFiles = []) => {
-			oauthConnectConfirmDialog(OCA.Mattermost.mattermostUrl).then((result) => {
-				if (result) {
-					if (OCA.Mattermost.usePopup) {
-						oauthConnect(OCA.Mattermost.mattermostUrl, OCA.Mattermost.clientId, null, true)
-							.then((data) => {
-								OCA.Mattermost.mattermostConnected = true
-								openChannelSelector(selectedFiles)
-							})
-					} else {
-						const selectedFilesIds = selectedFiles.map(f => f.id)
-						oauthConnect(
-							OCA.Mattermost.mattermostUrl,
-							OCA.Mattermost.clientId,
-							'files--' + OCA.Files.App.fileList._currentDirectory + '--' + selectedFilesIds.join(','),
-						)
-					}
-				}
-			})
-		},
+			return null
+		}).filter((e) => e !== null)
+		if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, after changeDirectory, files:', files)
+		if (files.length) {
+			if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, after changeDirectory, call openChannelSelector')
+			openChannelSelector(files)
+		}
 	}
+}
+
+function connectToMattermost(selectedFiles = []) {
+	oauthConnectConfirmDialog(OCA.Mattermost.mattermostUrl).then((result) => {
+		if (result) {
+			if (OCA.Mattermost.usePopup) {
+				oauthConnect(OCA.Mattermost.mattermostUrl, OCA.Mattermost.clientId, null, true)
+					.then((data) => {
+						OCA.Mattermost.mattermostConnected = true
+						openChannelSelector(selectedFiles)
+					})
+			} else {
+				const selectedFilesIds = selectedFiles.map(f => f.id)
+				const currentDirectory = OCA.Mattermost.currentFileList.folder.attributes.filename
```

Comment:
optional chaining after `folder` or maybe it is not required?

Reply:
:+1: